OpenKh.Command.IdxImg/Program.cs

```diff
@@ -213,6 +226,49 @@ protected int OnExecute(CommandLineApplication app)
 
                 using var isoStream = File.Open(InputIso, FileMode.Open, FileAccess.ReadWrite);
 
+                // Begin being hacky.
+
+                if (IdxIsoBlock == -1 || ImgIsoBlock == -1)
+                {
+                    byte[] imgNeedle = new byte[] { 0x01, 0x09, 0x4B, 0x48, 0x32, 0x2E, 0x49, 0x4D, 0x47, 0x3B, 0x31 };
+                    byte[] idxNeedle = new byte[] { 0x01, 0x09, 0x4B, 0x48, 0x32, 0x2E, 0x49, 0x44, 0x58, 0x3B, 0x31 };
+
+                    isoStream.Position = (uint)(0x105 * 0x800);
+                    byte[] haystack = isoStream.ReadBytes(0x500);
+
+                    using (BinaryReader reader = new BinaryReader(new MemoryStream(haystack)))
```

Comment:
I believe it is not required to read a `0x500` block and using it in the `BinaryReader`. Can you do `new BinaryReader(isoStream)` instead? If you are concerned about performance, you can wrap `isoStream` with a `BufferedStream`.